spec/filters/bytes_spec.rb

```diff
@@ -0,0 +1,208 @@
+
+    sample("32mb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024)
+    end
+
+    sample("32MB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024)
+    end
+  end
+
+  describe "from megabits" do
+    sample("32Mb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 / 8)
+    end
+  end
+
+  describe "from gigabytes" do
+    sample("32gB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024)
+    end
+
+    sample("32gb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024)
+    end
+
+    sample("32GB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024)
+    end
+  end
+
+  describe "from gigabits" do
+    sample("32Gb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 / 8)
+    end
+  end
+
+  describe "from terabytes" do
+    sample("32tB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32tb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32TB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024)
+    end
+  end
+
+  describe "from terabits" do
+    sample("32Tb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 / 8)
+    end
+  end
+
+  describe "from petabytes" do
+    sample("32pB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32pb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32PB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024)
+    end
+  end
+
+  describe "from petabits" do
+    sample("32Pb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024 / 8)
+    end
+  end
+
+  describe "from exabytes" do
+    sample("32eB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32eb") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024 * 1024)
+    end
+
+    sample("32EB") do
+      expect(subject).to include("dest")
+      expect(subject.get('dest')).to eq(32 * 1024 * 1024 * 1024 * 1024 * 1024 * 1024)
```

Comment:
This is a really edge case (today)...but found this a while back with AWS EFS ... it is possible for exabytes to overflow a long when represented as bytes. Maybe we shouldn't support exabytes ?

Reply:
Sure, that sounds safer. We can not support them until someone asks for them. Will remove. Thanks.